xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -379,16 +465,33 @@ private void updatePicker(
       }
 
       void shutdown() {
+        if (isDeactivated()) {
+          delayedDeletionTimer.cancel();
+          // not to set delayedDeletionTimer = null, in order to keep isDeactivated() unchanged
+        }
         childBalancer.shutdown();
       }
+
+      void reactivate() {
```

Comment:
This is reactivate, not activate. It does not make sense to activate when it's not previous deactivated, it will be no-op, and no-op is definitely no "activate".